build/toolchain/linux_arm/linux_arm_toolchain.gni

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/build.gni")
+
+import("${build_root}/config/android/config.gni")
+import("${build_root}/toolchain/gcc_toolchain.gni")
+
+declare_args() {
+  # similar to "CROSS_COMPILE" when build kernel
+  linux_arm_cross_compile = ""
+}
+
+template("linux_arm_toolchain") {
```

Comment:
There is nothing ARM specific about this. I think you are looking for just a customizable toolchain.